model-engine/model_engine_server/inference/vllm/vllm_server.py

```diff
@@ -1,89 +1,40 @@
 import asyncio
 import code
 import json
-import logging
 import os
 import signal
 import subprocess
 import traceback
 from logging import Logger
 from typing import AsyncGenerator, Dict, List, Optional
 
-import uvicorn
-from fastapi import BackgroundTasks, FastAPI, HTTPException, Request
-from fastapi.responses import JSONResponse, Response, StreamingResponse
-from vllm.engine.arg_utils import AsyncEngineArgs
-from vllm.engine.async_llm_engine import AsyncEngineDeadError, AsyncLLMEngine
+from fastapi import APIRouter, BackgroundTasks, HTTPException, Request
+from fastapi.responses import Response, StreamingResponse
+from vllm.engine.async_llm_engine import AsyncEngineDeadError
+from vllm.engine.protocol import AsyncEngineClient
+from vllm.entrypoints.launcher import serve_http
+from vllm.entrypoints.openai.api_server import build_async_engine_client, init_app
 from vllm.entrypoints.openai.cli_args import make_arg_parser
-from vllm.entrypoints.openai.protocol import ChatCompletionRequest as OpenAIChatCompletionRequest
-from vllm.entrypoints.openai.protocol import ChatCompletionResponse as OpenAIChatCompletionResponse
 from vllm.entrypoints.openai.protocol import CompletionRequest as OpenAICompletionRequest
-from vllm.entrypoints.openai.protocol import ErrorResponse
-from vllm.entrypoints.openai.serving_chat import OpenAIServingChat
-from vllm.entrypoints.openai.serving_completion import OpenAIServingCompletion
-from vllm.entrypoints.openai.serving_embedding import OpenAIServingEmbedding
 from vllm.model_executor.guided_decoding import get_guided_decoding_logits_processor
 from vllm.outputs import CompletionOutput
 from vllm.sampling_params import SamplingParams
 from vllm.sequence import Logprob
 from vllm.utils import FlexibleArgumentParser, random_uuid
 from vllm.version import __version__ as VLLM_VERSION
 
-logging.basicConfig(
-    format="%(asctime)s | %(levelname)s: %(message)s",
-    datefmt="%b/%d %H:%M:%S",
-    level=logging.INFO,
-)
-
 logger = Logger("vllm_server")
 
+async_engine_client: AsyncEngineClient
+
 TIMEOUT_KEEP_ALIVE = 5  # seconds.
 TIMEOUT_TO_PREVENT_DEADLOCK = 1  # seconds
-app = FastAPI()
-
-openai_serving_chat: OpenAIServingChat
-openai_serving_completion: OpenAIServingCompletion
-openai_serving_embedding: OpenAIServingEmbedding
-
-
-@app.get("/healthz")
-@app.get("/health")
-async def healthcheck():
-    await openai_serving_chat.engine.check_health()
-    return Response(status_code=200)
-
-
-@app.get("/v1/models")
-async def show_available_models():
-    models = await openai_serving_chat.show_available_models()
-    return JSONResponse(content=models.model_dump())
-
-
-@app.post("/v1/chat/completions")
-async def create_chat_completion(request: OpenAIChatCompletionRequest, raw_request: Request):
```

Comment:
guess we're gonna rethink these?

Reply:
I use `init_app` from vllm's openai entrypoint package which define these routes for us. I just added our /predict /stream routes

Reply:
oh got it, that makes sense